include/zenoh-pico/system/platform.h

```diff
@@ -30,6 +30,9 @@
 #error "Unknown platform"
 #endif
 
+/*------------------ Random ------------------*/
+uint8_t z_random(void);
```

Comment:
In standard C/C++ `rand()` is defined as `int rand(void)`. 
For the sake of compatibility, I believe it would be to expose the same function signature.
What do you think?

Reply:
I would agree with you at first, but then an `int` does not have the same size for all the platforms. Especially for small embedded devices on which an `int` can be 16-bits instead of 32-bits.

Because of that, and in order to have the same expected output across platforms, I opt for making it return a random value between 0-255. Bigger random values can still be generated by using something like this:
```
z_zint_t gen_zint(void)
{
    z_zint_t ret = 0;
    for (unsigned int i = 0; i < sizeof(z_zint_t); i++)
        ret |= z_random() << (i * 8);

    return ret;
}
```

Counter-proposal: To extend the z_random methods to:
```
int z_random(void);
uint8_t z_random_u8(void);
uint16_t z_random_u16(void);
uint32_t z_random_u32(void);
uint64_t z_random_u64(void);
z_zint_t z_random_zint(void);
```

what do you think?

Reply:
I believe it's a better proposal since it allows to get the best out of each platform. 
E.g.: ESP32 could make use of `esp_random` and `esp_fill_random` (see [here](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/api-reference/system/random.html)) to generate PRN instead of zenoh-pico building everything from a `uint8_t`.

Reply:
Addressed in the commit below.